packages/csharp/PortingAssistant/PortingAssistant.Telemetry/Utils/LogUploadUtils.cs

```diff
@@ -0,0 +1,213 @@
+            if (File.Exists(lastReadTokenFile))
+            {
+                fileLineNumberMap = JsonConvert.DeserializeObject<Dictionary<string, int>>(File.ReadAllText(lastReadTokenFile));
+            }
+            var initLineNumber = 0;            
+            foreach (var file in fileEntries) {
+              var fName = Path.GetFileNameWithoutExtension(file);
+              var fileExtension = Path.GetExtension(file);
+              var logName = prefix;
+
+              // Check which type of log file and set the prefix
+              if (fName == "main") {
+                continue;
+              } else {
+                string typeOfLog = fName.Split('-')[1];
+                if (typeOfLog == "assessment") {
+                  continue;
+                } else if (typeOfLog == "telemetry") {
+                  logName = "portingAssistant-metrics"; 
+                } else if (typeOfLog == "backend") {
+                  logName = "portingAssistant-backend-logs";
+                } else if (typeOfLog == "electron") {
+                  logName = "electron-logs";
+                } else if (typeOfLog == "react") {
+                  logName = "react-errors";
+                }
+              }
+              
+              var logs = new ArrayList();
+
+              // Add new files to fileLineNumberMap
+              if (!fileLineNumberMap.ContainsKey(file)) {
+                fileLineNumberMap[file] = 0;
+              }
+              initLineNumber = fileLineNumberMap[file];
+              FileInfo fileInfo = new FileInfo(file);
+              var success = false;
+              if (!IsFileLocked(fileInfo)) {
+                using (FileStream fs = fileInfo.Open(FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
+                {
+                    using (StreamReader reader = new StreamReader(fs))
+                    {
+                        string line = null;
+                        int currLineNumber = 0;
+                        for (; currLineNumber < initLineNumber; currLineNumber++)
+                        {
+                            line = reader.ReadLine();
+                            if (line == null)
+                            {
+                                return;
+                            }
+                        }
+
+                        line = reader.ReadLine();
+
+                        while (line != null)
+                        {
+                            currLineNumber++;
+                            logs.Add(line);
+                            line = reader.ReadLine();
+
+                            // send 1000 lines of logs each time when there are large files
+                            if (logs.Count >= 1000)
+                            {
+                                // logs.TrimToSize();
+                                success = PutLogData(client, logName, JsonConvert.SerializeObject(logs), profile, teleConfig).Result;
```

Comment:
If this fails, won't we keep appending the log without clearing it?

Reply:
Yes, and it persistently keeps trying to send all the logs. We can have a max cache limit for the logs and then start deleting them even if the api call fails or we can have a limit on number of lines.

Reply:
Yes, it keeps adding while still attempting to send all the logs. Since we have the logs in a log file we can try sending all the logs without missing anything once the put-log-api starts working.